parlai/core/torch_generator_agent.py

```diff
@@ -853,8 +852,15 @@ def advance(self, logprobs):
         if self.scores is None:
             self.scores = torch.zeros(1).type_as(logprobs).to(logprobs.device)
 
+        # penalize hypotheses ending in EOS on the prior scores (self.scores) level
+        # this is only related to search which uses prior scores (self.scores) (e.g. beam)
+        for hyp_id, token in enumerate(self.outputs[-1]):
+            if token == self.eos:
+                self.scores[hyp_id] = neginf(self.scores.dtype)
```

Comment:
correct me if i'm wrong: if you modified `logprobs[hyp_id]` here such that EVERYTHING was neginf except padding, then the assert I propose would be okay. (Nucleus and topk search would start only outputting padding)

Reply:
discussed offline another way